contrib/golang/upstreams/http/tcp/source/go/pkg/upstreams/http/tcp/api.h

```diff
@@ -0,0 +1,155 @@
+#pragma once
+
+// NOLINT(namespace-envoy)
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h> // NOLINT(modernize-deprecated-headers)
+
+typedef struct { // NOLINT(modernize-use-using)
+  const char* data;
+  uint64_t len;
+} Cstring;
+
+struct httpRequest;
+
+typedef struct { // NOLINT(modernize-use-using)
+  struct httpRequest* req;
+  int is_encoding;
+  int state;
+} processState;
+
+typedef struct httpRequest { // NOLINT(modernize-use-using)
+  Cstring plugin_name;
+  uint64_t configId;
+  // The ID of the worker that is processing this request, this enables the go filter to dedicate
+  // memory to each worker and not require locks
+  uint32_t worker_id;
+} httpRequest;
+
+typedef struct { // NOLINT(modernize-use-using)
+  uint64_t plugin_name_ptr;
+  uint64_t plugin_name_len;
+  uint64_t config_ptr;
+  uint64_t config_len;
+  int is_route_config;
```

Comment:
useless code?

Reply:
yes, is_route_config is useless. but other fields are actually useful:
1. plugin_name_ptr & plugin_name_len: in envoyGoOnTcpUpstreamConfig of go side, use them to  get related configParser of go so plugin.
2. config_ptr & config_len: in envoyGoOnTcpUpstreamConfig of go side, use them to get plugin_config.
![image](https://github.com/user-attachments/assets/5e67a0a5-8a2f-479a-b851-fa5541d42439)